### ISSUE TITLE
[Horovod on Spark] Error after  finishing last step of last epoch (Caught signal 11 - Segment fault: object not mapped to address)

### ISSUE THREAD

**Issue by wakoun** [horovod/horovod#3666]
**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet) : TensorFlow
2. Framework version: 2.2.0
3. Horovod version: 0.19.1
4. MPI version: 4.1.1
5. CUDA version: N/A
6. NCCL version: N/A
7. Python version: 3.7.11
8. Spark / PySpark version: 2.4.7
9. Ray version:
10. OS and version: Red Hat Enterprise Linux release 8.4
11. GCC version: 9.2.1
12. CMake version: 3.20.2

**Checklist:**
1. Did you search issues to find if somebody asked this question before? Yes but the solution did not help
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)? Yes

**Bug report:**
During training a model, at the last of the last epoch, these errors are shown:
[1,0]<stdout>:Epoch 3/3
[1,1]<stderr>:[datanode1:1071388:0:1071412] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x7f1d3a4d5e80)
[1,0]<stderr>:[datanode2:1074588:0:1074610] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x7f2bd5d6de80)
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
22/08/25 15:59:38 INFO scheduler.TaskSetManager: Finished task 0.0 in stage 13.0 (TID 31) in 511487 ms on datanode2 (executor 2) (1/2)
--------------------------------------------------------------------------
mpirun noticed that process rank 1 with PID 1071388 on node datanode1-e53c5cfa74cb78ea903ae2ece51b1b2d exited on signal 11 (Segmentation fault).
--------------------------------------------------------------------------

**Comment by chongxiaoc** [comment]
Do you have a reproducer, any existing Spark examples fail on your side?

**Comment by wakoun** [comment]
The error happened when running a deep learning workload on a horovod/spark environment. Working on a reproducer.

**Comment by Moriiitez** [comment]
Hey , I have gotten into the same issue as well, in my case some of the worker nodes get error " Caught signal 11 (Segmentation fault: address not mapped to object at address " and others get "Error in sys.excepthook:"

have you been able to solve the issue ?
